### DISCUSSION TITLE
Key-based authentication

### DISCUSSION THREAD

**Discussion by asyncrom** [serverpod/serverpod#1995] [Q&A] [answered]
Hi 👋
I'm creating an app that doesn't require storing massive / important data.
To make it simpler for my users, i would like to make a key-based authentication:
   - When downloading the app an unique id is created, and identifies the user.
   - If the user want to log in the same account the next time, he can just save his identifier on his own, and use it to log in.

Is there any way to implement this with serverpod, so i can take benefits from the Session and User utils, without using the email authentication provided ?
Thanks in advance if anyone answers !
Asyncrom 🐧

**Comment by Frank1016** [comment, marked as answer]
You can do it like this: when the unique id is created, create the user email and password automatically (here the email can be any string as long as it's unique and follows email format) linked to this unique id. These info together should be saved somewhere for the auto login next time.

**Reply by asyncrom** [reply]
Wow, i hadn't thought about that, thanks !
I guess i need to find a way ti disable email verification then.

**Reply by Frank1016** [reply]
No need to disable the verification, you can also make that process automatic and the user doesn't need to get involved at all

**Reply by asyncrom** [reply]
How can i do that, the Auth package works only with emails and Google/Apple auth. Using email, there is a code i need to send to the user, is there a way to bypass this step ?

**Comment by Frank1016** [comment]
The verification code actually is saved in one of the serverpod tables. It's possible to extract it from there, but I haven't tried doing this yet. Maybe one of the serverpod devs can share some instructions here.

**Reply by asyncrom** [reply]
I have managed to do it by putting the code the Redis cache, it works perfectly.
